modules/metrics/src/main/java/com/spotify/apollo/metrics/MetricsModule.java

```diff
@@ -127,12 +127,13 @@ public SemanticMetricRegistry semanticMetricRegistry() {
   @Provides
   @Singleton
   public MetricsFactory apolloMetrics(
-      SemanticMetricRegistry metricRegistry, MetricsConfig metricsConfig
+      SemanticMetricRegistry metricRegistry, MetricsConfig metricsConfig, Config config
   ) {
     return new SemanticMetricsFactory(
         metricRegistry,
         what -> metricsConfig.serverMetrics().contains(what),
-        metricsConfig.precreateCodes()
+        metricsConfig.precreateCodes(),
+        config
```

Comment:
Why did you choose to pass in the entire `Config` object instead of adding the methods you need to the `MetricsConfig`?